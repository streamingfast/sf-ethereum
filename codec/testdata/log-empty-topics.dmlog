FIRE INIT 2.4 geth 1.13.15-fh2.4-db154a58-2024041
FIRE BEGIN_BLOCK 3325385
FIRE BEGIN_APPLY_TRX 7bc3bbdb67ac4b66f3f4aaa28b23c0e4138f3aa8a8d9ec42733a4f2ef04153b2 5efaafa76a873d16060c1804ebc90319463f34d7 . 01 dab0262d044a390d1f005b767fb89db5bb6625d5b7b5be9137c429bd741a2f95 46885e294a4b8bd0f22eb192a011bcd538f94c6c28a935f6ec5af62f4f549c8b 3000000 9502f907 271 e82ea46c000000000000000000000000698bb513889287e2007b780f4abaf3e4de639445000000000000000000000000000000000000000000000000000000000000000c 00 9502f90e 9502f900 2 1 3 0 . .
FIRE TRX_FROM 1a79cd22b49bc83ca571bd079cfb542bd85578ce
FIRE EVM_RUN_CALL CALL 1 5
FIRE EVM_PARAM CALL 1 1a79cd22b49bc83ca571bd079cfb542bd85578ce 5efaafa76a873d16060c1804ebc90319463f34d7 . 2978440 e82ea46c000000000000000000000000698bb513889287e2007b780f4abaf3e4de639445000000000000000000000000000000000000000000000000000000000000000c
FIRE ADD_LOG 1 0 5efaafa76a873d16060c1804ebc90319463f34d7  3d602d80600a3d3981f3363d3d373d3d3d363d73698bb513889287e2007b780f4abaf3e4de6394455af43d8280 7
FIRE EVM_RUN_CALL CREATE 2 10
FIRE EVM_PARAM CREATE 2 5efaafa76a873d16060c1804ebc90319463f34d7 2761a281b541833e62dfbda44c42dad93d85a048 . 2899011 .
FIRE EVM_RUN_CALL DELEGATE 3 17
FIRE EVM_PARAM DELEGATE 3 2761a281b541833e62dfbda44c42dad93d85a048 698bb513889287e2007b780f4abaf3e4de639445 . 2851131 .
FIRE EVM_END_CALL 3 2851131 . 18
FIRE EVM_END_CALL 2 2896354 . 22
FIRE EVM_CALL_FAILED 1 2942243 execution reverted
FIRE EVM_REVERTED 1
FIRE EVM_END_CALL 1 2942243 . 24
FIRE END_APPLY_TRX 57757 . 120757 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 27 0 . []
FIRE FINALIZE_BLOCK 3325385
FIRE END_BLOCK 3325385 3575 {"finalizedBlockHash":"0x7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109","finalizedBlockNum":"0x32bdc9","header":{"parentHash":"0x8eededb3ef47265acb8bd4faeca40dbc0b19773667a9cc86ea45f7113e2c459f","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","miner":"0x9a6034c84cd431409ac1a35278c7da36ffda53e5","stateRoot":"0x7ee881ba39d4664ab5743b8575664bba86b76683aae9b4dd5ccc605dfe0c67f5","transactionsRoot":"0xd3f5a653a01d19315b1534125eb627a19aaf8fc60c8dc86f2e111f67aca447ca","receiptsRoot":"0x3c3d95195f9ea78b1ee5c7104db0713c8b5a0c72e98f0fe03dd49f48a551f7fe","logsBloom":"0x04200000004000008000000400000000000080000000000040000040000000080000000000000000000080000000060000001000000000000400000800040800001000000000000000800008000000000000000000048004000040000000000000000000020000000000800002000800000000020000400000000010000000082200100000000000000000002804000001000400000008008044000000000000008000000000000000000500000000002002000000002000000000004400400000000002000180000000000000000000010000000000000000000080020060000200000000000000000000000000000000210000008000000000000000000000","difficulty":"0x0","number":"0x32bdc9","gasLimit":"0x1c9c380","gasUsed":"0xb0614","timestamp":"0x6441081c","extraData":"0xd883010b05846765746888676f312e32302e32856c696e7578","mixHash":"0x3b19b0099cb461b256793c359ec1037079703e45460788d02b4e1cbae079546b","nonce":"0x0000000000000000","baseFeePerGas":"0x7","withdrawalsRoot":"0x769f72989ccf56ae46c1be2117dc41847a02bfc1e5eefd5f740f92def5fa257b","blobGasUsed":null,"excessBlobGas":null,"parentBeaconBlockRoot":null,"hash":"0x7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109"},"totalDifficulty":"0x3c656d23029ab0","uncles":[]}
