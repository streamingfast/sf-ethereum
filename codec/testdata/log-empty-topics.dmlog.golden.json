[
  {
    "hash": "7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109",
    "header": {
      "baseFeePerGas": {
        "bytes": "07"
      },
      "coinbase": "9a6034c84cd431409ac1a35278c7da36ffda53e5",
      "difficulty": {
        "bytes": "00"
      },
      "extraData": "d883010b05846765746888676f312e32302e32856c696e7578",
      "gasLimit": "30000000",
      "gasUsed": "722452",
      "hash": "7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109",
      "logsBloom": "04200000004000008000000400000000000080000000000040000040000000080000000000000000000080000000060000001000000000000400000800040800001000000000000000800008000000000000000000048004000040000000000000000000020000000000800002000800000000020000400000000010000000082200100000000000000000002804000001000400000008008044000000000000008000000000000000000500000000002002000000002000000000004400400000000002000180000000000000000000010000000000000000000080020060000200000000000000000000000000000000210000008000000000000000000000",
      "mixHash": "3b19b0099cb461b256793c359ec1037079703e45460788d02b4e1cbae079546b",
      "number": "3325385",
      "parentHash": "8eededb3ef47265acb8bd4faeca40dbc0b19773667a9cc86ea45f7113e2c459f",
      "receiptRoot": "3c3d95195f9ea78b1ee5c7104db0713c8b5a0c72e98f0fe03dd49f48a551f7fe",
      "stateRoot": "7ee881ba39d4664ab5743b8575664bba86b76683aae9b4dd5ccc605dfe0c67f5",
      "timestamp": "2023-04-20T09:38:36Z",
      "totalDifficulty": {
        "bytes": "3c656d23029ab0"
      },
      "transactionsRoot": "d3f5a653a01d19315b1534125eb627a19aaf8fc60c8dc86f2e111f67aca447ca",
      "uncleHash": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
      "withdrawalsRoot": "769f72989ccf56ae46c1be2117dc41847a02bfc1e5eefd5f740f92def5fa257b"
    },
    "number": "3325385",
    "size": "3575",
    "transactionTraces": [
      {
        "beginOrdinal": "1",
        "calls": [
          {
            "address": "5efaafa76a873d16060c1804ebc90319463f34d7",
            "callType": "CALL",
            "caller": "1a79cd22b49bc83ca571bd079cfb542bd85578ce",
            "endOrdinal": "24",
            "executedCode": true,
            "failureReason": "execution reverted",
            "gasConsumed": "36197",
            "gasLimit": "2978440",
            "index": 1,
            "input": "e82ea46c000000000000000000000000698bb513889287e2007b780f4abaf3e4de639445000000000000000000000000000000000000000000000000000000000000000c",
            "logs": [
              {
                "address": "5efaafa76a873d16060c1804ebc90319463f34d7",
                "data": "3d602d80600a3d3981f3363d3d373d3d3d363d73698bb513889287e2007b780f4abaf3e4de6394455af43d8280",
                "ordinal": "7",
                "topics": [
                  ""
                ]
              }
            ],
            "stateReverted": true,
            "statusFailed": true,
            "statusReverted": true
          },
          {
            "address": "2761a281b541833e62dfbda44c42dad93d85a048",
            "beginOrdinal": "10",
            "callType": "CREATE",
            "caller": "5efaafa76a873d16060c1804ebc90319463f34d7",
            "depth": 1,
            "endOrdinal": "22",
            "gasConsumed": "2657",
            "gasLimit": "2899011",
            "index": 2,
            "parentIndex": 1,
            "stateReverted": true
          },
          {
            "address": "698bb513889287e2007b780f4abaf3e4de639445",
            "beginOrdinal": "17",
            "callType": "DELEGATE",
            "caller": "2761a281b541833e62dfbda44c42dad93d85a048",
            "depth": 2,
            "endOrdinal": "18",
            "gasLimit": "2851131",
            "index": 3,
            "parentIndex": 2,
            "stateReverted": true
          }
        ],
        "endOrdinal": "27",
        "from": "1a79cd22b49bc83ca571bd079cfb542bd85578ce",
        "gasLimit": "3000000",
        "gasPrice": {
          "bytes": "9502f907"
        },
        "gasUsed": "57757",
        "hash": "7bc3bbdb67ac4b66f3f4aaa28b23c0e4138f3aa8a8d9ec42733a4f2ef04153b2",
        "index": 3,
        "input": "e82ea46c000000000000000000000000698bb513889287e2007b780f4abaf3e4de639445000000000000000000000000000000000000000000000000000000000000000c",
        "maxFeePerGas": {
          "bytes": "9502f90e"
        },
        "maxPriorityFeePerGas": {
          "bytes": "9502f900"
        },
        "nonce": "271",
        "r": "dab0262d044a390d1f005b767fb89db5bb6625d5b7b5be9137c429bd741a2f95",
        "receipt": {
          "cumulativeGasUsed": "120757",
          "logsBloom": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
        },
        "s": "46885e294a4b8bd0f22eb192a011bcd538f94c6c28a935f6ec5af62f4f549c8b",
        "status": "REVERTED",
        "to": "5efaafa76a873d16060c1804ebc90319463f34d7",
        "type": "TRX_TYPE_DYNAMIC_FEE",
        "v": "01"
      }
    ],
    "ver": 3
  }
]